datalad/distribution/subdatasets.py

```diff
@@ -6,364 +6,8 @@
-            # first deletions
-            for dprop in assure_list(delete_property):
-                parser.remove_option(submodule_section, dprop)
-                # also kick from the info we just read above
-                sm.pop('gitmodule_{}'.format(dprop), None)
-            # and now setting values
-            for sprop in assure_list(set_property):
-                prop, val = sprop
-                if val.startswith('<') and val.endswith('>') and '{' in val:
-                    # expand template string
-                    val = val[1:-1].format(
-                        **dict(
-                            sm,
-                            refds_relpath=relpath(sm['path'], refds_path),
-                            refds_relname=relpath(sm['path'], refds_path).replace(os.sep, '-')))
-                parser.set_value(
-                    submodule_section,
-                    prop,
-                    val)
-                # also add to the info we just read above
-                sm['gitmodule_{}'.format(prop)] = val
-            Dataset(dspath).save(
-                '.gitmodules', to_git=True,
-                message='[DATALAD] modified subdataset properties')
-            # let go of resources, locks, ...
-            parser.release()
-
-        #common = commonprefix((with_pathsep(subds), with_pathsep(path)))
-        #if common.endswith(sep) and common == with_pathsep(subds):
-        #    candidates.append(common)
-        subdsres = get_status_dict(
-            'subdataset',
-            status='ok',
-            type='dataset',
-            logger=lgr)
-        subdsres.update(sm)
-        subdsres['parentds'] = dspath
-        if not bottomup and \
-                (fulfilled is None or
-                 GitRepo.is_valid_repo(sm['path']) == fulfilled):
-            yield subdsres
-
-        # expand list with child submodules. keep all paths relative to parent
-        # and convert jointly at the end
-        if recursive and \
-                (recursion_limit in (None, 'existing') or
-                 (isinstance(recursion_limit, int) and
-                  recursion_limit > 1)):
-            for r in _get_submodules(
-                    sm['path'],
-                    fulfilled, recursive,
-                    (recursion_limit - 1)
-                    if isinstance(recursion_limit, int)
-                    else recursion_limit,
-                    contains,
-                    bottomup,
-                    set_property,
-                    delete_property,
-                    refds_path):
-                yield r
-        if bottomup and \
-                (fulfilled is None or
-                 GitRepo.is_valid_repo(sm['path']) == fulfilled):
-            yield subdsres
-    if parser is not None:
-        # release parser lock manually, auto-cleanup is not reliable in PY3
-        parser.release()
+# Import command class to ease 3rd-party transitions
```

Comment:
Should we issue a warning (probably with warnings.warn() rather than logging.warn()) when this module is imported?

Reply:
I don't think it is needed. Even in our own code base we did not import it directly.

Reply:
OK.